packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/MrwEnumSerializationProvider.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Generator.CSharp.Expressions;
+using static Microsoft.Generator.CSharp.Expressions.Snippets;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    internal class MrwEnumSerializationProvider : TypeProvider
+    {
+        private readonly EnumTypeProvider _enumType;
+
+        public MrwEnumSerializationProvider(EnumTypeProvider enumType) : base(null)
+        {
+            _enumType = enumType;
+            Namespace = _enumType.Namespace;
```

Comment:
this is taking the namespace of the generated enum type not the input enum type so that the extension here would be in the same namespace as the enum type.